### ISSUE TITLE
Subdivx not working

### ISSUE THREAD

**Issue by santiagomartinez** [morpheus65535/bazarr#2714]
Hi all!
I just saw this issue: https://github.com/morpheus65535/bazarr/issues/2702
But in my case even after downloading tag [v1.4.6-beta.2](https://github.com/morpheus65535/bazarr/releases/tag/v1.4.6-beta.2) I'm seeing in the logs:

```2024-10-12 01:49:51|INFO    |subliminal_patch.core           |Listing subtitles with provider 'subdivx' and languages {<Language: alpha3=spa;country=MX>}|
2024-10-12 01:49:51|DEBUG   |subliminal_patch.providers.subdivx|Titles to look at: ['Requiem for a Dream', '夢之安魂曲', '梦的挽歌', '레퀴엠', 'Реквієм по мрії']|
2024-10-12 01:49:51|DEBUG   |urllib3.connectionpool          |https://www.subdivx.com:443 "GET /inc/gt.php?gt=1 HTTP/1.1" 200 None|
2024-10-12 01:49:51|DEBUG   |subliminal_patch.providers.subdivx|Query: Requiem for a Dream|
2024-10-12 01:49:51|DEBUG   |urllib3.connectionpool          |https://www.subdivx.com:443 "POST /inc/ajax.php HTTP/1.1" 200 None|
2024-10-12 01:49:51|DEBUG   |subliminal_patch.providers.subdivx|JSONDecodeError: b''|
2024-10-12 01:49:51|DEBUG   |urllib3.connectionpool          |https://www.subdivx.com:443 "GET /inc/gt.php?gt=1 HTTP/1.1" 200 None|
2024-10-12 01:49:51|DEBUG   |subliminal_patch.providers.subdivx|Query: Requiem for a Dream (2000)|
2024-10-12 01:49:51|DEBUG   |urllib3.connectionpool          |https://www.subdivx.com:443 "POST /inc/ajax.php HTTP/1.1" 200 None|
2024-10-12 01:49:51|DEBUG   |subliminal_patch.providers.subdivx|JSONDecodeError: b''|
```
Can anyone please confirm it is still working and it is on my side or it is not working again?

Thank you very much!

**Comment by morpheus65535** [comment]
Could it be a temporary bug (returning an empty response) since it seems to work for other users now.

**Comment by santiagomartinez** [comment]
It seems to be down right now. I'll let you know with new tests results after it returns. Thanks!

**Comment by santiagomartinez** [comment]
It seems to be returning empty. I tested it with curl. I don't understand why because I compare it to the request payload seen in the browser where it works ok and both payloads match.

Token: 
`curl -X GET "https://subdivx.com/inc/gt.php?gt=1" -H "Content-Type: application/json"`

Search: 
```
curl -X POST "https://subdivx.com/inc/ajax.php" \
-H "Content-Type: application/json" \
-d '{"tabla": "resultados", "filtros": "", "buscar395a": "inside out", "token": "<TOKEN_VALUE>"}'
 ```

**Comment by morpheus65535** [comment]
Could it be that they implemented Cloudflare protection?

**Comment by alvarochvz** [comment]
![image](https://github.com/user-attachments/assets/d1fa8b19-ee60-461f-aee6-2fd38ac1b8eb)


Apparently

**Comment by morpheus65535** [comment]
I can confirm they've implemented Cloudflare protection. I won't fight against them. If someone's interested, I'll accept a PR.

**Comment by alvarochvz** [comment]
Is this worth looking into?
https://trash-guides.info/Prowlarr/prowlarr-setup-flaresolverr/

I have never used it honestly i am not sure how well it will playout with bazarr 🤔 

Is it even worth looking into?

**Comment by morpheus65535** [comment]
I've already tried to implement it without success but it may worth trying again.

**Comment by pombocaesar** [comment]
None of the other providers for Spanish are working correctly, it would be greatly appreciated if you could fix this one. Cheers

**Comment by LanusFac** [comment]
I’m having the same problem here. Bazarr just stopped working out of the blue when using SubDivx.

Has anyone managed to find the answer?

**Comment by morpheus65535** [comment]
> I’m having the same problem here. Bazarr just stopped working out of the blue when using SubDivx.
> 
> 
> 
> Has anyone managed to find the answer?

There's no fix.

**Comment by pombocaesar** [comment]
> > I’m having the same problem here. Bazarr just stopped working out of the blue when using SubDivx.
> > Has anyone managed to find the answer?
> 
> There's no fix.

A couple weeks ago you mentioned trying without luck and it would probably worth another shot (I assume after this issue gaining popularity).
Does your last message mean you already retried and couldn't or that you are certain through some new information that it definitely can't be achieved?

**Comment by anderson-oki** [comment]
@pombocaesar 

Putting in simple words, the provider implemented protection against non-humans to prevent automation. You can't simply bypass that. And we respect that they are protecting against robots.

**Comment by morpheus65535** [comment]
I agree with @anderson-oki. To better answer your question, I haven't retried and I din't have time to implement such feature for now.

**Comment by anderson-oki** [comment]
Just in case someone wants to invest time, i did a very quick check, and it seems that subdivx is not using the subliminal_patch http session that uses cloudscraper and it is simply using the `requests`.

One could try the luck with cloudscraper.

**Comment by morpheus65535** [comment]
> Just in case someone wants to invest time, i did a very quick check, and it seems that subdivx is not using the subliminal_patch http session that uses cloudscraper and it is simply using the `requests`.
> 
> 
> 
> One could try the luck with cloudscraper.

I tried with no luck.

**Comment by dvillaseca** [comment]
Hi, after doing some testing, they are using the most basic detection with cloudflare on the get token request, if you use the cloudscraper library it can by bypassed

**Comment by morpheus65535** [comment]
@dvillaseca yeah won't be that easy. Even if I use RetryingCFSession() instead of Session(), I end up with a session token but /inc/ajax.php return a 403. Probably something from the cookiejar I must bring with me. I'll have to dig a little deeper but I'll take your input if you have some. :-)

**Comment by dvillaseca** [comment]
> @dvillaseca yeah won't be that easy. Even if I use RetryingCFSession() instead of Session(), I end up with a session token but /inc/ajax.php return a 403. Probably something from the cookiejar I must bring with me. I'll have to dig a little deeper but I'll take your input if you have some. :-)

I managed to fix it and make it work but i'm a bit scare to share it, looks like they went to big depths in order to block automation, we maybe have to respect their effort and wait to see if they release a vip subscription system for automation like opensubtitles

**Comment by morpheus65535** [comment]
> I managed to fix it and make it work but i'm a bit scare to share it, looks like they went to big depths in order to block automation, we maybe have to respect their effort and wait to see if they release a vip subscription system for automation like opensubtitles

It's up to you. I wont fight against them forever, I'll fix this once and if they continue to implement new security, I'll drop the towel and respect their choice.

**Comment by Airle** [comment]
This is a message shared by the Subdivx developer:

**"Due to abuses caused by automated software, which overload the server, I had to implement some strategies to mitigate its impact."**

There is no mention of any kind of VIP subscription.

**Comment by morpheus65535** [comment]
@Airle where did you see that message? Any way to contact site admins?

**Comment by Airle** [comment]
> @Airle where did you see that message? Any way to contact site admins?

He left that message on his subdivx profile. His username is "braytac," so I guess you can contact him directly there. Look for him under "Miembros"

It also seems he has an account here: https://github.com/braytac.

**Comment by dvillaseca** [comment]
nice, let's contact him and ask for a vip feature

**Comment by morpheus65535** [comment]
I don't have an account on subdivx so it will be hard to achieve on my side.

**Comment by pombocaesar** [comment]
I do and I'm also Argentinian (he is, based on .ar in his account TLD) so
I'll reach out to him now.

**Comment by morpheus65535** [comment]
> I do and I'm also Argentinian (he is, based on .ar in his account TLD) so I'll reach out to him now.

Thanks, tell him we don't want to fight and we understand and respect his choice. We'll be happy to discuss with him on Discord.

**Comment by alvarochvz** [comment]
Hey so if there is a possibility of this being a premium feature let us know -- i would also be on board if you need hands to code such feature.

**Comment by braytac** [comment]
> I can confirm they've implemented Cloudflare protection. I won't fight against them. If someone's interested, I'll accept a PR.

Greetings!
I am a developer of subdivx (not the owner).
One of the things I had proposed 2 years ago was to make an API to work with subliminal, but I still didn't have time.
I had even less time in the last few months, because half the time I had to build different strategies to mitigate an extremely abusive consumption of the server. I had the mysqld process at 100% all the time. I had load average at 30!!
First I started to filter for requests that came from the "subliminal" agent (exactly the one I wanted to run, since it is the one I use the most). But after a while the consumption came back and was much worse.
No, we don't have any protection from cloudflare. I simply implemented a penalty strategy for each query identified as abusive. The more retries there are in a period of time defined as short, the waiting time is multiplied x2, without returning results.
That way, I was able to get the server back to a normal load average of ~1.5, where everything works fine for users who download manually from the web.
I have no problem coordinating with you any special request dedicated to automated software, which is less expensive for the DB engine, and so that there are no successive requests in a very short period of time.
For movies, there shouldn't be much of a problem. For series, you could also adapt to iterate on a single response in json, for a given season. Having the IDs of each subtitle already, you wouldn't need to query the database again for each specific chapter.

**Comment by dvillaseca** [comment]
@morpheus65535 hello, have you coordinated something like this in the past?
I'm not completely familiarized on how bazarr queries each provider when searching for tv show subtitles, 
I have seen that bazarr makes tons for search queries, there is a way of reducing the amount of queries made so we don't overload the server?

**Comment by morpheus65535** [comment]
@braytac sorry for the delay responding to your message. Of course we can work together to reduce the footprint of Bazarr on your website. Here's some suggestions:
- We stop using random user-agent and provide something that you can catch and work with for throttling or blocking.
- We make sure to impose a delay between queries (5 seconds?) to limit number of queries sent by a single user.
- If you implement x-rate-limit headers, we can follow what they instruct.
- Without going down the full API path, if you can provide 2 GET endpoints that we could call using imdb number and, for series, season and episode numbers, we could use this and simply get json back in response. This would prevent a lot of js, css, jpeg and other stuff being downloaded which is totally useless.
- We could cache answer for non-existant movies or series and prevent multiple queries for the same stuff from reaching your server within 24 hours.

Anything that make sense to you here? If you want to discuss further, easiest way may be on our Discord dev-community channel.

Thanks!

**Comment by santiagomartinez** [comment]
I don't know what changed but it started to work again some hours ago:

<img width="480" alt="Image" src="https://github.com/user-attachments/assets/5a0931b9-4771-48f7-af90-7e1f464e802d" />

**Comment by santiagomartinez** [comment]
For now fixed it adding a sleep before each query and removing alternative title queries, including the query without the year for movies. With that only a single query is performed for each title.

**Comment by dvillaseca** [comment]
yep i did something similar, it also helps if you login with a divx
account, the throttling is reduced

**Comment by morpheus65535** [comment]
@braytac I haven't heard from you since my last post. I'm still available if you want to work on this with me!

**Comment by braytac** [comment]
Greetings!

Sorry for the delay. I did this. 

If we see that it does not overload the server too much, we will increase (or decrease) the number of results.
The quota per user is currently 100 search queries, with LIMIT 100 records for each of them. There is no limit for the number of downloads. I think the quota is more generous than the one opensubtitles has.

Usage:
 <TOOL> "https://subdivx.com/api/?<parameter1>&<parameter2>&..." 

Description:
 This command retrieves data from the Subdivx API. The API requires specific
 parameters to process your request and return the desired information.
 This request should be a GET request, meaning the parameters are sent
 directly in the URL.

Required Parameters:
 user: Your Subdivx username.
 pass: Your Subdivx password.
 search: The title of the movie or TV show you are searching for.
 api_key: 'MFiHrGP87Tkkca9zGMvXRxFZgHAHleJNbJGII2orBWM=' 
            (shh!! xD)

Optional Parameter:
 filter: (Optional) A filter to refine your search results. This could be
 used to specify a particular release group, version, or quality.

Example:

curl "https://subdivx.com/api/?user=myuser&pass=mypassword&search=Star+trek&filter=YTS.MX&api_key=<token>"

Download:
https://subdivx.com/descargar.php?id=<ID>

Notes:
 - Replace the placeholders (<USER>, <PASS>, <TITLE>, <RIPPER/VERSION>, <token>)
 with your current values.
 - The 'filter' parameter is optional. If not provided, the API will return
 all matching results.

**Comment by sebigut** [comment]
@morpheus65535 sorry if its wrong to tag you, I was wondering if you have been able to see brayac response.

Thank you for all the work, bazarr is awesome

**Comment by morpheus65535** [comment]
@sebigut yes I did, sorry, I'm way too much overwhelmed with work and family right now. I'll rework this provider when I get some free time.

**Comment by morpheus65535** [comment]
> Sorry for the delay. I did this.

My turn for being sorry for the delay! I'm trying to implement your API in replacement for parsing your web pages. It doesn't seems to be possible to create an account right now, am I right?

Some additional questions:
- the 100 search limits: is it daily? Can I get the remaining search through response headers or something similar? I would like to throttle before hammering you with useless queries.
- can we search using movie imdb_id or series imdb_id? It would be nice instead of having to do multiple queries with different titles (English or localized titles).
- For series, can we specify season and episode numbers?
- May be useless in your case but can we specify the desired languages? Hearing-impaired? Forced/foreign?

Thanks!

**Comment by sebigut** [comment]
@morpheus65535 I can lend you my account if neccesary.
I tried on my own the API endpoint and can confirm that the search parameter only accepts titles (at least right now).

![Image](https://github.com/user-attachments/assets/d7b45ff6-c697-46f5-bee3-c914a5a5595e)

Also this example with the filter on:

![Image](https://github.com/user-attachments/assets/cfa67d46-1614-4c5c-9a21-d336dcea122d)

**Comment by dvillaseca** [comment]
> Greetings!
> 
> Sorry for the delay. I did this.
> 
> If we see that it does not overload the server too much, we will increase (or decrease) the number of results. The quota per user is currently 100 search queries, with LIMIT 100 records for each of them. There is no limit for the number of downloads. I think the quota is more generous than the one opensubtitles has.
> 
> Usage: `<TOOL> "https://subdivx.com/api/?<parameter1>&<parameter2>&..."`
> 
> Description: This command retrieves data from the Subdivx API. The API requires specific parameters to process your request and return the desired information. This request should be a GET request, meaning the parameters are sent directly in the URL.
> 
> Required Parameters: user: Your Subdivx username. pass: Your Subdivx password. search: The title of the movie or TV show you are searching for. api_key: 'MFiHrGP87Tkkca9zGMvXRxFZgHAHleJNbJGII2orBWM=' (shh!! xD)
> 
> Optional Parameter: filter: (Optional) A filter to refine your search results. This could be used to specify a particular release group, version, or quality.
> 
> Example: `curl "https://subdivx.com/api/?user=<USER>&pass=<PASS>&search=Star+trek&filter=YTS.MX&api_key=<token>"`
> 
> Download: `https://subdivx.com/descargar.php?id=<ID>`
> 
> Notes:
> 
> * The 'filter' parameter is optional. Applies to the subtitle details/description field.
>   If not provided, the API will return
>   all matching results.

Hello, 
Is it possible for the api to return the description on the response? 
It would be a big help because if not, we would need to request the description individually for each result leading to higher server load

Thanks!

**Comment by Airle** [comment]
any news?

**Comment by morpheus65535** [comment]
> I can lend you my account if neccesary.

I wouldn't want to put you into trouble. If you have a way to reach the site admin, let him know I would appreciate if he could provide me a test account.

**Comment by JMVS** [comment]
Maybe this helps? https://api.subt.is/v1/docs
It's a meta-search engine that includes SubDivX

EDIT: Github repo: https://github.com/lndgalante/subtis

**Comment by lndgalante** [comment]
Hey @JMVS I'm the developer of Subtis, please let me know how can I help to integrate Bazarr with Subtis

**Comment by JMVS** [comment]
> Hey [@JMVS](https://github.com/JMVS) I'm the developer of Subtis, please let me know how can I help to integrate Bazarr with Subtis

Hey! Thanks for popping in! You can get started on how to integrate it here: https://wiki.bazarr.media/Troubleshooting/FAQ/#im-missing-or-want-a-subtitle-provider-added-to-bazarr

**Comment by Airle** [comment]
Any updates on this?

**Comment by lndgalante** [comment]
I haven't time yet to build the Bazarr + Subtis integration, but is on our roadmap. If anyone want to help building it, I can help on guidance on how to use the Subtis API is pretty straightforward.

**Comment by braytac** [comment]
News in API search
- "filter" can be multiple: filter=<patterrn> or filter=pattern1,pattern3,pattern3...
- If search matches with an imdb ID format, then search by IMDB. Else, normal search by text.

by example:

https://subdivx.com/api/?user=&pass=&search=Star+trek+2009&filter=EZTV&api_key=...

https://subdivx.com/api/?user=&pass=&search=tt0091209&filter=YTS,algo1,algo2,algo3....&api_key=...

**Comment by morpheus65535** [comment]
@braytac without an account, there's not much I can do to improve this provider. Any way to provide me one?

**Comment by braytac** [comment]
> [@braytac](https://github.com/braytac) without an account, there's not much I can do to improve this provider. Any way to provide me one?

Hello! Sorry for the delay. Done :)

I'll take this opportunity to share two new features:
From a few days now, the exact content of compressed files has also been (and will be) extracted. Since the translator usually retains the original name paired with the video file, you can now perform an exact search for the SRT file that matches the MP4/MKV. Just use the video (or subtitle) file name as a search pattern. It will detect that it has an extension and infer that it's an exact search (unless the extension is missing).

By example:
 wget "https://subdivx.com/api/?user=bazarr&pass=......&search=It 2017 720p BluRay x264 - [YTS.AG].mkv&api_key=........=" -O - | cat

**Comment by rcordobar19** [comment]
Hey there,

I was able to bypass cloudflare and retrieve subtitle data, even download it using a flaresolverr container (that I already have on my homelab for prowlarr) via postman.

I'm gonna write a PR, to connect it all in a couple of days, in the meantime this is how I did it:

Using flaresolverr I queried subdivx's token endpoint `https://subdivx.com/inc/gt.php?gt=1` this will return the `userAgent` used and the `cf_clearance` cookie which is essential to make the rest of the requests directly to the subdivx endpoints.

Then you just need to grab those cookies, the token and the userAgent and craft the search request:
POST - https://subdivx.com/inc/ajax.php
Cookie: sdx=1234; cf_clearance=12345;
User-Agent: userAgent from flaresolverr
(not sure if below headers are necessary at all)
content-type: application/x-www-form-urlencoded; charset=UTF-8
accept: application/json, text/javascript, */*; q=0.01
x-requested-with: XMLHttpRequest

Body: tabla=resultados&filtros=&buscar421=Robin+Hood+2025+S01E06&token=TOKEN

and that's it, youll get a list of subtitles.
I dont know how feasible will it be to add flaresolverr support to bazarr and this is my first time poking around into this code so any recommendations are welcome.

**Comment by lndgalante** [comment]
Hey everyone, Leo from [Subtis](https://subtis.io/), we know there's a big demand for Bazarr support, and among those lines we just launched Kodi support, therefore now we're putting all of our efforts and energy on the Bazarr integration. I'll share more updates here soon!

**Comment by lndgalante** [comment]
Some good news 🙌 We've a PR up to integrate [Subtis](https://subtis.io/) plugin for Bazarr, please go ahead and react to it with some emojis, also if you have any comments please feel free to drop them in that PR.

PR 👉 https://github.com/morpheus65535/bazarr/pull/3114

> Hey everyone, Leo from [Subtis](https://subtis.io/), we know there's a big demand for Bazarr support, and among those lines we just launched Kodi support, therefore now we're putting all of our efforts and energy on the Bazarr integration. I'll share more updates here soon!

**Comment by lndgalante** [comment]
> Some good news 🙌 We've a PR up to integrate [Subtis](https://subtis.io/) plugin for Bazarr, please go ahead and react to it with some emojis, also if you have any comments please feel free to drop them in that PR.
> 
> PR 👉 [#3114](https://github.com/morpheus65535/bazarr/pull/3114)
> 
> > Hey everyone, Leo from [Subtis](https://subtis.io/), we know there's a big demand for Bazarr support, and among those lines we just launched Kodi support, therefore now we're putting all of our efforts and energy on the Bazarr integration. I'll share more updates here soon!

PR updated, please give it a thumbs up!

https://github.com/morpheus65535/bazarr/pull/3114

**Comment by lvitti** [comment]
I’m leaving another alternative to get Subdivx subtitles that you might want to take into account.

https://subx-api.duckdns.org/docs

**Comment by lndgalante** [comment]
What's that provider @lvitti ? Seems broken to me, and the homepage has a Clerk login.


<img width="4140" height="2800" alt="Image" src="https://github.com/user-attachments/assets/f52c2498-4207-4418-9db8-b15b07421074" />

<img width="4140" height="2800" alt="Image" src="https://github.com/user-attachments/assets/5913c6b7-b6ec-47b4-ac79-96ab8964f833" />

**Comment by sleighton** [comment]
It's a really strange landing page. Try this instead  https://subx-api.duckdns.org/docs/getting-started/introduction/

**Comment by lvitti** [comment]
It’s just an API wrapper for subdivx. It handles all the searching, downloading, and Cloudflare-related stuff.
The docs seem to be working for me. That's not the best doc page, but the API works fine, and the login/register is only to generate your own API key.

**Comment by lndgalante** [comment]
That's great, is that SubX project from you @sleighton and @lvitti ?

**Comment by sleighton** [comment]
@lndgalante No, I'm just a Bazarr user following this thread, I literally know nothing about SubX.

**Comment by morpheus65535** [comment]
> PR updated, please give it a thumbs up!

PR merged! Will be part of upcoming beta!

**Comment by lvitti** [comment]
@lndgalante Yes, the project is mine. I created it to support my plugins for [emby](https://github.com/lvitti/emby-SubdivX)/[jellyfin](https://github.com/lvitti/jellyfin-plugin-subdivx) and to avoid extra dependencies when searching on subdivx

**Comment by lndgalante** [comment]
@lvitti that's amazing! I just integrated SubX into Subtis indexer, let me know if we can support or contribute in any way, shape or form to SubX since the project is pretty critical !

**Comment by lvitti** [comment]
Thanks, @lndgalante ! I just opened this discussion board: https://github.com/lvitti/subx/discussions to have a place to discuss issues or feature.

**Comment by PuyGonzalo** [comment]
Hi @lndgalante, Is there anything need to be configured in order to use provider Subtis? It's not working ok, lots of subtitles not found even if I know they are in SubDivx. And it's only working for movies and not for shows.

**Comment by lndgalante** [comment]
@PuyGonzalo we've indexed so far ~9k subtitles, our goal is to have only subtitles at the highest quality that 100% matches the user movie version.  Currently we're prioritizing search accuracy over subtitles volume at this point.

Anyways we will have a new indexer built in Go soon to index faster and more subtitles.

**Comment by santiagomartinez** [comment]
> I’m leaving another alternative to get Subdivx subtitles that you might want to take into account.
> 
> https://subx-api.duckdns.org/docs

For some reason I lost more than 400 subtitles. I've running nightly without realizing for some time, maybe some unstable version did something unexpected at some point so I had to find a way of searching for all of them again. I tried with the Flaresolverr PR but the searches didn't work.
Finally I adapted the current Subdivx provider to use Subx API and it works great. Thank you very much @lvitti, nice work <3.

**Comment by PuyGonzalo** [comment]
Hello @santiagomartinez,

> Finally I adapted the current Subdivx provider to use Subx API and it works great. Thank you very much @lvitti, nice work <3.

Would it be possible for you to share how you did this? I can't figure it out.

**Comment by santiagomartinez** [comment]
@PuyGonzalo https://gist.github.com/santiagomartinez/a0ffbd473e514372a235b81d8877f668

Replace `custom_libs/subliminal_patch/providers/subdivx.py` with that.
Also, replace "XXXXXX" with your API key. It is hardcoded as I didn't change the form on the providers selection on the UI.

It has a 5 secs sleep after each request to avoid hitting the API too frequently

**Comment by JMVS** [comment]
I think Subdivx provider should be discarded in favor of Subtis and SubX

**Comment by morpheus65535** [comment]
I'm closing this issue since this provider isn't supported anymore and alternatives have raised.

**Comment by Treat-me-kindly** [comment]
@morpheus65535 SubX GitHub is no longer visible, and Subtis, despite using Subdivx as one of the sources, is nowhere close to Subdix. @santiagomartinez shared a good fix above :(

**Comment by morpheus65535** [comment]
@Treat-me-kindly subdivx is no long part of the supported subtitles providers list in Bazarr.

**Comment by santiagomartinez** [comment]
> [@morpheus65535](https://github.com/morpheus65535) SubX GitHub is no longer visible, and Subtis, despite using Subdivx as one of the sources, is nowhere close to Subdix. [@santiagomartinez](https://github.com/santiagomartinez) shared a good fix above :(

My solution used SubX which is now integrated into Bazarr.
In SubX the search by the parameter `imdb_id` doesn't seem to always work. Example:
`
/api/subtitles/search?limit=200&video_type=movie&imdb_id=tt9620288&year=2021
`
returns 0 results, but
`
/api/subtitles/search?limit=200&video_type=movie&title=King%20Richard&year=2021
`
results are ok. This may require a new issue.

**Comment by lvitti** [comment]
> > [@morpheus65535](https://github.com/morpheus65535) SubX GitHub is no longer visible, and Subtis, despite using Subdivx as one of the sources, is nowhere close to Subdix. [@santiagomartinez](https://github.com/santiagomartinez) shared a good fix above :(
> 
> My solution used SubX which is now integrated into Bazarr. In SubX the search by the parameter `imdb_id` doesn't seem to always work. Example: `/api/subtitles/search?limit=200&video_type=movie&imdb_id=tt9620288&year=2021` returns 0 results, but `/api/subtitles/search?limit=200&video_type=movie&title=King%20Richard&year=2021` results are ok. This may require a new issue.

I’m the creator of Subx. subx only uses the data provided by subdivx.

Step by step I try to make improvements and adjustments to generate more acceptable results, while maintaining the essence of subdivx.

For example, the case you mentioned has the same issue on subdivx. If you search by IMDb, it returns no results because the subtitles are not associated with that IMDb entry.

I invite you to visit the repository https://github.com/lvitti/subx/issues to leave any issues or suggestions you may have. I will try to resolve them as soon as possible.

**Comment by santiagomartinez** [comment]
> > > [@morpheus65535](https://github.com/morpheus65535) SubX GitHub is no longer visible, and Subtis, despite using Subdivx as one of the sources, is nowhere close to Subdix. [@santiagomartinez](https://github.com/santiagomartinez) shared a good fix above :(
> > 
> > 
> > My solution used SubX which is now integrated into Bazarr. In SubX the search by the parameter `imdb_id` doesn't seem to always work. Example: `/api/subtitles/search?limit=200&video_type=movie&imdb_id=tt9620288&year=2021` returns 0 results, but `/api/subtitles/search?limit=200&video_type=movie&title=King%20Richard&year=2021` results are ok. This may require a new issue.
> 
> I’m the creator of Subx. subx only uses the data provided by subdivx.
> 
> Step by step I try to make improvements and adjustments to generate more acceptable results, while maintaining the essence of subdivx.
> 
> For example, the case you mentioned has the same issue on subdivx. If you search by IMDb, it returns no results because the subtitles are not associated with that IMDb entry.
> 
> I invite you to visit the repository https://github.com/lvitti/subx/issues to leave any issues or suggestions you may have. I will try to resolve them as soon as possible.

Thanks for the answer. Indeed I'll post the issue in there, but my opinion is that in this case this should be handled by bazarr subx provider and not your api.

**Comment by JMVS** [comment]
> > > > [@morpheus65535](https://github.com/morpheus65535) SubX GitHub is no longer visible, and Subtis, despite using Subdivx as one of the sources, is nowhere close to Subdix. [@santiagomartinez](https://github.com/santiagomartinez) shared a good fix above :(
> > > 
> > > 
> > > My solution used SubX which is now integrated into Bazarr. In SubX the search by the parameter `imdb_id` doesn't seem to always work. Example: `/api/subtitles/search?limit=200&video_type=movie&imdb_id=tt9620288&year=2021` returns 0 results, but `/api/subtitles/search?limit=200&video_type=movie&title=King%20Richard&year=2021` results are ok. This may require a new issue.
> > 
> > 
> > I’m the creator of Subx. subx only uses the data provided by subdivx.
> > Step by step I try to make improvements and adjustments to generate more acceptable results, while maintaining the essence of subdivx.
> > For example, the case you mentioned has the same issue on subdivx. If you search by IMDb, it returns no results because the subtitles are not associated with that IMDb entry.
> > I invite you to visit the repository https://github.com/lvitti/subx/issues to leave any issues or suggestions you may have. I will try to resolve them as soon as possible.
> 
> Thanks for the answer. Indeed I'll post the issue in there, but my opinion is that in this case this should be handled by bazarr subx provider and not your api.

I disagree. This should be taken care from the closest thing to the source, if not the source itself. The integrations should be dealing with bugs within themselves and not external sources, even if the integration uses that source. I could see this if this was the only option available due to lack of support or EoL.
